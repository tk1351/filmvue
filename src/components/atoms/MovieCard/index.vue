<script setup lang="ts">
withDefaults(
  defineProps<{
    poster_path?: string | null
    original_title?: string
    release_date?: string
  }>(),
  {
    poster_path: '',
    original_title: '',
    release_date: '',
  }
)

interface Emits {
  (eventName: 'click:link', event: Event): void
}
const emit = defineEmits<Emits>()

const handleLinkClick = (event: Event) => {
  event.preventDefault()
  emit('click:link', event)
}
</script>

<template>
  <section class="movie-card">
    <header>
      <a data-test-id="img-link" @click="handleLinkClick">
        <img class="movie-card__img" :alt="original_title" :src="poster_path" />
      </a>
    </header>
    <div class="movie-card__content">
      <h3 class="movie-card__title">
        <a data-test-id="title-link" @click="handleLinkClick">{{
          original_title
        }}</a>
      </h3>
      <p class="movie-card__date">{{ release_date }}</p>
    </div>
  </section>
</template>

<style scoped lang="scss">
.movie-card {
  width: 300px;
  height: 350px;
  border: 1px solid $main-color;
  border-radius: 30px;
  display: flex;
  flex-direction: column;
}

.movie-card__img {
  width: 100%;
  height: 220px;
  border-radius: 30px 30px 0 0;

  &:hover {
    opacity: 0.8;
  }
}

.movie-card__content {
  padding: 14px 20px;
  border-radius: 0 0 30px 30px;
  background-color: $main-color;
  flex-grow: 1;
}

.movie-card__title {
  margin-bottom: 14px;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;

  &:hover {
    opacity: 0.8;
  }
}
</style>
