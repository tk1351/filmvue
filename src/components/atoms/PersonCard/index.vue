<script setup lang="ts">
withDefaults(
  defineProps<{
    profile_path?: string | null
    name?: string
    department?: string
  }>(),
  {
    profile_path: '',
    name: '',
    department: undefined,
  }
)

interface Emits {
  (eventName: 'click:link', event: Event): void
}
const emit = defineEmits<Emits>()

const handleLinkClick = (event: Event) => {
  event.preventDefault()
  emit('click:link', event)
}
</script>

<template>
  <section class="person-card">
    <header>
      <a data-test-id="img-link" @click="handleLinkClick">
        <img class="person-card__img" :src="profile_path" :alt="name" />
      </a>
    </header>
    <div class="person-card__content">
      <h3 class="person-card__name">
        <a data-test-id="name-link" @click="handleLinkClick">{{ name }}</a>
      </h3>
      <p v-if="department">{{ department }}</p>
    </div>
  </section>
</template>

<style scoped lang="scss">
.person-card {
  width: 200px;
  height: 330px;
  border: 1px solid $main-color;
  border-radius: 30px;
  display: flex;
  flex-direction: column;
}

.person-card__img {
  width: 100%;
  height: 220px;
  border-radius: 30px 30px 0 0;

  &:hover {
    opacity: 0.8;
    cursor: pointer;
  }
}

.person-card__content {
  padding: 14px 20px;
  border-radius: 0 0 30px 30px;
  background-color: $main-color;
  flex-grow: 1;
}

.person-card__name {
  margin-bottom: 14px;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;

  &:hover {
    opacity: 0.8;
    cursor: pointer;
  }
}
</style>
