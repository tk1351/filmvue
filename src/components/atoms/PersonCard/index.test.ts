import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import PersonCard from './index.vue'
import MovieCard from '../MovieCard/index.vue'

describe('Testing PersonCard Component', () => {
  it('PersonCardコンポーネントが正しく表示されること', () => {
    const wrapper = mount(PersonCard)
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('Props', () => {
    describe('profile_path', () => {
      it('imgタグのsrc属性へ、正しく値が渡っていること', () => {
        const expectedSrc = 'expectedSrc'
        const wrapper = mount(PersonCard, {
          props: {
            profile_path: expectedSrc,
          },
        })
        const image = wrapper.find('img')
        expect(image.attributes('src')).toBe(expectedSrc)
      })
    })

    describe('name', () => {
      it('imgのalt属性へ、正しく値が渡っていること', () => {
        const expectedName = 'expectedName'
        const wrapper = mount(PersonCard, {
          props: {
            name: expectedName,
          },
        })
        const image = wrapper.find('img')
        expect(image.attributes('alt')).toBe(expectedName)
      })
      it('h3タグの中に、正しく値が渡っていること', () => {
        const expectedName = 'expectedName'
        const wrapper = mount(PersonCard, {
          props: {
            name: expectedName,
          },
        })
        expect(wrapper.find('h3').text()).toContain(expectedName)
      })
    })
    describe('department', () => {
      it('pタグに、正しく値が渡っていること', () => {
        const expectedDepartment = 'expectedDepartment'
        const wrapper = mount(PersonCard, {
          props: {
            department: expectedDepartment,
          },
        })
        expect(wrapper.find('p').text()).toContain(expectedDepartment)
      })
    })
  })

  describe('Event', () => {
    it('画像のリンクをクリックした時、click:linkイベントが1度実行されること', async () => {
      const wrapper = mount(PersonCard)
      await wrapper.find("[data-test-id='img-link']").trigger('click')
      expect(wrapper.emitted('click:link')).toHaveLength(1)
    })
    it('名前のリンクをクリックした時、click:linkイベントが1度実行されること', async () => {
      const wrapper = mount(PersonCard)
      await wrapper.find("[data-test-id='name-link']").trigger('click')
      expect(wrapper.emitted('click:link')).toHaveLength(1)
    })
  })
})
