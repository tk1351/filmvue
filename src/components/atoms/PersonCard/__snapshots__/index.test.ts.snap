// Vitest Snapshot v1

exports[`Testing PersonCard Component > PersonCardコンポーネントが正しく表示されること 1`] = `
<section
  class="person-card"
  data-v-1bfbd1b6=""
>
  <header
    data-v-1bfbd1b6=""
  >
    <a
      data-test-id="img-link"
      data-v-1bfbd1b6=""
    >
      <img
        alt=""
        class="person-card__img"
        data-v-1bfbd1b6=""
        src=""
      />
    </a>
  </header>
  <div
    class="person-card__content"
    data-v-1bfbd1b6=""
  >
    <h3
      class="person-card__name"
      data-v-1bfbd1b6=""
    >
      <a
        data-test-id="name-link"
        data-v-1bfbd1b6=""
      />
    </h3>
    <!--v-if-->
  </div>
</section>
`;
